lib/Driver/DarwinToolChains.cpp

```diff
@@ -221,9 +221,46 @@ static bool wantsObjCRuntime(const llvm::Triple &triple) {
   llvm_unreachable("unknown Darwin OS");
 }
 
+/// Return the earliest backward deployment compatibility version we need to
+/// link in for the given target triple, if any.
+static Optional<std::pair<unsigned, unsigned>>
```

Comment:
Nitpick: can you use llvm::VersionTuple for this? Bigger, sure, but also obviously a version. (And has an "empty" representation, though the Optional is fine too.)